async/deferred.ts

```diff
@@ -0,0 +1,27 @@
+// Copyright 2018-2019 the Deno authors. All rights reserved. MIT license.
```

Comment:
I don't know about `/async` ... seems a bit too specific.

I feel like `/util` would be a useful junk drawer for one-off things like this. 
@hayd what do you think?

Reply:
I also considered to let it `promise` FYI. I don't have any special reason to `async`

Reply:
Can you rename this to `//util/deferred.ts` ?